src/miniscript/astelem.rs

```diff
@@ -554,7 +574,12 @@ where
                     .map(|sub| expression::terminal(sub, Pk::from_str))
                     .collect();
 
-                pks.map(|pks| Terminal::Multi(k, pks))
+                if frag_name == "multi" {
+                    pks.map(|pks| Terminal::Multi(k, pks))
+                } else {
+                    // must be multi_a
+                    pks.map(|pks| Terminal::MultiA(k, pks))
+                }
```

Comment:
I do believe this should be `match`; otherwise with future TapLeaf version-defined post-TapScripts we will not have a compiler error here

Reply:
There is a match check that frag_name is either `multi` or `multi_a`. I don't see an issue with doing an if-else here. 

```rust
("multi", n) | ("multi_a", n) => {
  				// code..
                if frag_name == "multi" {
                    pks.map(|pks| Terminal::Multi(k, pks))
                } else {
                    // must be multi_a
                    pks.map(|pks| Terminal::MultiA(k, pks))
                }
            }
```